src/App.tsx

```diff
@@ -1,73 +1,108 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+
+interface Person {
+  name: string;
+  born: number;
+  died: number;
+  sex: string;
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selected, setSelected] = useState(false);
+  const [selectedPerson, setSelectedPerson] =
+    useState<string>('No selected person');
+
+  const applyQuery = () => {
+    debounce(() => setSelected(true), 300)();
+  };
+
+  const handleQueryChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setSelectedPerson('No selected person');
+  };
+
+  const filteredPeople = useMemo(() => {
+    const normalisedQuery = query.toLowerCase().trim();
+
+    return peopleFromServer.filter(person => {
+      const normalisedPerson = person.name.toLowerCase();
+
+      return normalisedPerson.includes(normalisedQuery);
+    });
+  }, [query]);
+
+  const handlePersonClick = (person: Person) => {
+    setQuery(person.name);
+    setSelected(false);
+    setSelectedPerson(() => {
+      return `${person.name} (${person.born} - ${person.died})`;
+    });
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson}
         </h1>
 
         <div className="dropdown is-active">
           <div className="dropdown-trigger">
             <input
               type="text"
               placeholder="Enter a part of the name"
+              value={query}
               className="input"
               data-cy="search-input"
+              onChange={handleQueryChanged}
+              onFocus={applyQuery}
             />
           </div>
 
           <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
```

Comment:
it is said in the task that you should have delay as a prop, so probably you need to move dropdown logic to a custom component